pow-migration/src/lib.rs

```diff
@@ -80,6 +81,52 @@ pub fn get_block_windows(network_id: NetworkId) -> Result<&'static BlockWindows,
     }
 }
 
+/// Report as online (if necessary)
+/// Returns the last block number in which the online status was reported
+pub async fn report_online(
+    pow_client: &Client,
+    block_windows: &BlockWindows,
+    pow_block_number: u32,
+    validator_address: Address,
+) -> Result<u32, Error> {
+    // We calculate the range of the online transactions, which start one readiness window (1 day) before the election candidate.
+    let online_start = block_windows.election_candidate - block_windows.readiness_window;
+    let online_range = online_start..block_windows.election_candidate;
+
+    // First we obtain the list of online txns that we have sent
+    let txns = get_online_txns(pow_client, validator_address.to_string(), online_range).await;
+
+    let transaction = generate_online_tx(validator_address.to_user_friendly_address());
+
+    // If we have not reported ready yet, we do so:
+    if txns.is_empty() {
+        log::info!("Reporting as online for the first time...");
+        send_tx(pow_client, transaction).await?;
+        return Ok(pow_block_number);
+    } else {
+        // Get the latest online transaction that was sent
+        let mut latest_online_bn = 0;
+        for txn in txns {
+            if txn.block_number > latest_online_bn {
+                latest_online_bn = txn.block_number;
+            }
+        }
```

Comment:
get_online_txns is already calling `.iter()...collect()`. It is only ever used here as far as I can tell and thus it could be changed to 
```rust
/// Checks if we have seen an online transaction from a validator in the specified range
pub async fn get_last_online(
    pow_client: &Client,
    validator: String,
    block_window: Range<u32>,
) -> Option<u32> {
```
or similar performing this check.

Reply:
I would leave `get_online_txns` as it is because I can imagine the use case of using this function to dump or print the list of online txns for a given validator. Similar to the "list stakers" command